public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,380 @@
+  const styles = useStyles2(getStyles);
+
+  // Filter state
+  const [eventsFilter, setEventsFilter] = useState('');
+  // form for filter fields
+  const { register, handleSubmit, reset } = useForm({ defaultValues: { query: '' } }); //  form for search field
+  const from = timeRange?.from.unix();
+  const to = timeRange?.to.unix();
+  const onFilterCleared = useCallback(() => {
+    setEventsFilter('');
+    reset();
+  }, [setEventsFilter, reset]);
+
+  const {
+    data: stateHistory,
+    isLoading,
+    isError,
+    error,
+  } = stateHistoryApi.endpoints.getRuleHistory.useQuery(
+    {
+      from: from,
+      to: to,
+      limit: LIMIT_EVENTS,
+    },
+    {
+      refetchOnFocus: true,
+      refetchOnReconnect: true,
+    }
+  );
+
+  const { historyRecords } = useRuleHistoryRecords(stateHistory, eventsFilter);
+
+  if (isError) {
+    return <HistoryErrorMessage error={error} />;
+  }
+
+  return (
+    <Stack direction="column" gap={1}>
+      <div className={styles.labelsFilter}>
+        <form onSubmit={handleSubmit((data) => setEventsFilter(data.query))}>
+          <SearchFieldInput
+            {...register('query')}
+            showClearFilterSuffix={!!eventsFilter}
+            onClearFilterClick={onFilterCleared}
+          />
+          <input type="submit" hidden />
+        </form>
+      </div>
+      <LoadingIndicator visible={isLoading} />
+      <HistoryLogEvents logRecords={historyRecords} />
+    </Stack>
+  );
+};
+
+// todo: this function has been copied from RuleList.v2.tsx, should be moved to a shared location
+const LoadingIndicator = ({ visible = false }) => {
+  const [measureRef, { width }] = useMeasure<HTMLDivElement>();
+  return <div ref={measureRef}>{visible && <LoadingBar width={width} />}</div>;
+};
+
+interface HistoryLogEventsProps {
+  logRecords: LogRecord[];
+}
+function HistoryLogEvents({ logRecords }: HistoryLogEventsProps) {
+  // display log records
+  return (
+    <ul>
+      {logRecords.map((record) => {
+        return <EventRow key={record.timestamp + hashLabelsOrAnnotations(record.line.labels ?? {})} record={record} />;
+      })}
+    </ul>
+  );
+}
+
+interface HistoryErrorMessageProps {
+  error: unknown;
+}
+
+function HistoryErrorMessage({ error }: HistoryErrorMessageProps) {
+  if (isFetchError(error) && error.status === 404) {
+    return <EntityNotFound entity="History" />;
+  }
+  const title = t('central-alert-history.error', 'Something went wrong loading the alert state history');
+
+  return <Alert title={title}>{stringifyErrorLike(error)}</Alert>;
+}
+
+interface SearchFieldInputProps {
+  showClearFilterSuffix: boolean;
+  onClearFilterClick: () => void;
+}
+const SearchFieldInput = React.forwardRef<HTMLInputElement, SearchFieldInputProps>(
+  ({ showClearFilterSuffix, onClearFilterClick, ...rest }: SearchFieldInputProps, ref) => {
+    const placeholder = t('central-alert-history.filter.placeholder', 'Filter events in the list by labels');
```

Comment:
Phrasing: this would be either:

* "Filter events in the list by label" (singular "label", even though they might use more than one)
* or "Filter events in the list with labels"

The first is a little more idiomatic & natural-sounding to my ear. Maybe the second is better for people who don't have English as a first language.

Reply:
Thanks Rob! I will update the text with your suggestion. With the final PR we will ping Brenda for reviewing all the texts in the feature. Not pinging her yet , as these might change